### PR TITLE
Add codeclimate to our pipeline

### PR DESCRIPTION
**Pull request by miah**
The goal with this PR is to add 'codeclimate' to our pipeline. This means all PR's would get checked for complexity and maintainability. We can tune its reports through the included yaml file and I suspect that at first it might be brutal. =)

https://codeclimate.com/github/inspec/inspec

https://github.com/inspec/inspec/pull/3988

Signed-off-by: Miah Johnson <miah@chia-pet.org>

### CODE REVIEW

File: .codeclimate.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+version: "2"
+checks:
+  file-lines:
+    enabled: false
+plugins:
+  fixme:
+    enabled: true
+    config:
+      strings:
+        - TODO
```
Comment (zenspider) [inline .codeclimate.yml:10]: I also use HACK, FIX, REFACTOR... ummm... 

But I also don't think I've added any either (yet)

File: .codeclimate.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  flog:
+    enabled: true
```
Comment (zenspider) [inline .codeclimate.yml:12]: Flay? (I think it is called `duplication`)

That one is gonna be brutal as well. We might want to bump thresholds to start and slowly ramp them down over time.

File: .codeclimate.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+version: "2"
+checks:
+  file-lines:
+    enabled: false
```
Comment (zenspider) [inline .codeclimate.yml:4]: I honestly wish this wasn't disabled... I like to code on a fairly small laptop and it's been a bit of a pain.

File: .codeclimate.yml (hunk 2 above)
Comment (zenspider) [reply .codeclimate.yml:12]: It might also be this:

```
  similar-code:
    enabled: true
  identical-code:
    enabled: true
```

**Review by zenspider — APPROVED** [review]
I think this is a good start. It's prolly gonna hurt. (yay!)

File: .codeclimate.yml (hunk 2 above)
Comment (zenspider) [reply .codeclimate.yml:12]: Confirmed that it has been renamed: https://docs.codeclimate.com/docs/duplication-concept

File: .codeclimate.yml (hunk 1 above)
Comment (miah) [reply .codeclimate.yml:10]: I did some quick greps and it seems like TODO was the only one present in our repo. I haven't looked at train, kitchen-inspec, etc so this list might grow/change soon. (I'll likely end up copying this config to all our repos in the future).

File: .codeclimate.yml (hunk 2 above)
Comment (miah) [reply .codeclimate.yml:12]: Correct we should get those automatically as they are enabled by default. I'm only disabling the defaults and adding optional plugins in this configuration.

Comment (miah) [reply .codeclimate.yml:12]: Do you have any suggestions for a good starting threshold for flog?

File: .codeclimate.yml (hunk 3 above)
Comment (miah) [reply .codeclimate.yml:4]: I'm in agreement. I'll gladly remove this entry, I know our code severely fails on this one though.

@clintoncwolfe whats your thought?

Comment (clintoncwolfe) [reply .codeclimate.yml:4]: Well, philosophically, I'm the "I want to build features and fix bugs on a product, not deal with linting issues constantly", so I don't care if the files are big, etc. We have editors, we have scrolling mice, we have search.

On new work, I'm all for short files, why not; but I don't see much gain from reducing file length in isolation from other refactorings (I'm all for splitting lib/inspec/cli.rb, for example, but it's not file-length that motivates me).

That is entirely an opinion, though.

Comment (miah) [reply .codeclimate.yml:4]: Lets leave it disabled for now =)

Comment (zenspider) [reply .codeclimate.yml:4]: Oh. Derp. I assumed it was about line width, not file length.

**Comment by aaronlippold** [comment]
Can I also suggest a bit of static analysis as well.

On going static analysis That's part of the regular PR process would go a
long way in demonstrating a focus on security as well as function.

File: .codeclimate.yml (hunk 2 above)
Comment (zenspider) [reply .codeclimate.yml:12]: One MILLION!

**Pull request merged**